### ISSUE TITLE
[Flex] Cache does not rebuild on new YAML file

### ISSUE THREAD

**Issue by weaverryan** [symfony/symfony#25904]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4 (really, it's a Flex thing)

To reproduce:

1) Install a bundle that does NOT create a config file for you (which is totally fine if you bundle doesn't normally need configuration). For example: `composer require knplabs/knp-markdown-bundle`

2) Create a config file - e.g. `config/packages/knp_markdown.yaml`.

The container will *not* rebuild automatically. In `Kernel.php`, we're loading some `glob` via the `GlobResource`, but it doesn't appear that those `GlobResource` are *actually* set into the container and thus considered on refresh. On container build, I see about 12 `GlobResource` being created. But on refresh, I only see `isFresh()` being called on 2 of them (both looking in `src/` or `/src/Controller` for changes) - I also only see those 2 in the .meta file.

It *appears* like we're creating `GlobResource` in `FileLoader`, using them to find files, but never adding them to the container as resources. Or am I mistaken?

Thanks!

**Comment by stof** [comment]
that's weird. We do have this call here: https://github.com/symfony/symfony/blob/332ad0ab9bd5a24b7f86435277346a0ad3ddd019/src/Symfony/Component/DependencyInjection/Loader/FileLoader.php#L154
so it is meant to work.

**Comment by weaverryan** [comment]
@stof That's the mechanism for the PSR4 service loading stuff, which *is* working. But in this case, I believe the `glob()` method is called - https://github.com/symfony/symfony/blob/332ad0ab9bd5a24b7f86435277346a0ad3ddd019/src/Symfony/Component/Config/Loader/FileLoader.php#L123 - and THIS is never added to the container. But I could be over-simplifying things.

**Comment by weaverryan** [comment]
The pending question is this:

Should the glob import that's used in `Kernel.php`:

```php
$loader->load($confDir.'/packages/*'.self::CONFIG_EXTS, 'glob');
```

cause a resource to be added to the container so that new files matching this glob cause a cache rebuild? Or is the code in `Kernel.php` incorrect, and we need more code there to make sure new files are watched for?

**Comment by nicolas-grekas** [comment]
Should be fixed by #25989
